File: app/src/main/java/com/arman/queuetube/activities/PlaylistActivity.java
package com.arman.queuetube.activities;

import android.os.Bundle;
import android.view.MenuItem;

import com.arman.queuetube.config.Constants;
import com.arman.queuetube.fragments.PlaylistFragment;

import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

public class PlaylistActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.setupActionBar();

        String playlistName = getIntent().getStringExtra(Constants.Fragment.Argument.PLAYLIST_NAME);
        Bundle bundle = new Bundle();
        bundle.putString(Constants.Fragment.Argument.PLAYLIST_NAME, playlistName);
        PlaylistFragment fragment = new PlaylistFragment();
        fragment.setArguments(bundle);

        getSupportFragmentManager().beginTransaction().replace(android.R.id.content, fragment).commit();
    }

    private void setupActionBar() {
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        switch (id) {
            case android.R.id.home:
                finish();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/java/com/arman/queuetube/activities/PlaylistActivity.kt
package com.arman.queuetube.activities

import android.os.Bundle
import com.arman.queuetube.config.Constants
import com.arman.queuetube.fragments.playlist.PlaylistFragment

class PlaylistActivity : BackActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val playlistName = intent.getStringExtra(Constants.Fragment.Argument.PLAYLIST_NAME)
        val isEditable = intent.getBooleanExtra(Constants.Fragment.Argument.IS_EDITABLE, true)
        val isShufflable = intent.getBooleanExtra(Constants.Fragment.Argument.IS_SHUFFLABLE, true)
        val isSortable = intent.getBooleanExtra(Constants.Fragment.Argument.IS_SORTABLE, true)
        val isDraggable = intent.getBooleanExtra(Constants.Fragment.Argument.IS_DRAGGABLE, true)
        val bundle = Bundle()
        bundle.putString(Constants.Fragment.Argument.PLAYLIST_NAME, playlistName)
        bundle.putBoolean(Constants.Fragment.Argument.IS_EDITABLE, isEditable)
        bundle.putBoolean(Constants.Fragment.Argument.IS_SHUFFLABLE, isShufflable)
        bundle.putBoolean(Constants.Fragment.Argument.IS_SORTABLE, isSortable)
        bundle.putBoolean(Constants.Fragment.Argument.IS_DRAGGABLE, isDraggable)
        val fragment = PlaylistFragment()
        fragment.arguments = bundle

        supportFragmentManager.beginTransaction().replace(android.R.id.content, fragment).commit()
    }

}
